src/main/java/seedu/address/storage/JsonSerializableJobFestGo.java

```diff
@@ -96,12 +97,16 @@ public JobFestGo toModelType() throws IllegalValueException {
         }
         for (JsonAdaptedTask jsonAdaptedTask : taskList) {
             Task task = jsonAdaptedTask.toModelType();
-            task = new Task(task.getDescription(), task.getDate(),
-                    jobFestGo.getEvent(task.getAssociatedEventName()), task.isCompleted());
-            if (jobFestGo.hasTask(task)) {
-                throw new IllegalValueException(MESSAGE_DUPLICATE_TASK);
+            try {
+                task = new Task(task.getDescription(), task.getDate(),
+                        jobFestGo.getEvent(task.getAssociatedEventName()), task.isCompleted());
+                if (jobFestGo.hasTask(task)) {
+                    throw new IllegalValueException(MESSAGE_DUPLICATE_TASK);
+                }
+                jobFestGo.addTask(task);
+            } catch (EventNotFoundException enfe) {
+                throw new IllegalValueException("Event not found!");
```

Comment:
perhaps can store this error message in the form of a static variable to follow the codebase so something like `MESSAGE_EVENT_NOT_FOUND`

Reply:
Good suggestion! Have adjusted accordingly.